mcs/class/System/System.Net.Sockets/SafeSocketHandle.cs

```diff
@@ -48,8 +55,17 @@ protected override bool ReleaseHandle ()
 					int abort_attempts = 0;
 					while (blocking_threads.Count > 0) {
 						if (abort_attempts++ >= ABORT_RETRIES) {
-							if (THROW_ON_ABORT_RETRIES)
-								throw new Exception ("Could not abort registered blocking threads before closing socket.");
+							if (THROW_ON_ABORT_RETRIES) {
+								StringBuilder sb = new StringBuilder ();
+								sb.AppendLine ("Could not abort registered blocking threads before closing socket.");
+								foreach (var thread in blocking_threads) {
+									sb.AppendLine ("Thread StackTrace:");
```

Comment:
Adding Thread.Name/Thread.ManagedThreadId here would probably make it even better.